src/Groups/group_constructors.jl

```diff
@@ -127,6 +127,27 @@ i.e., if `G` can be generated by one element.
 """
 @gapattribute is_cyclic(G::GAPGroup) = GAP.Globals.IsCyclic(G.X)::Bool
 
+"""
+    cyclic_generator(G::GAPGroup)
+
+Return an element of `G` that generates `G` if `G` is cyclic,
+and throw an error otherwise.
+
+# Examples
+```jldoctest
+julia> g = permutation_group(5, [cperm(1:3), cperm(4:5)])
+Group([ (1,2,3), (4,5) ])
+
+julia> cyclic_generator(g)
+(1,2,3)(4,5)
+```
+"""
+function cyclic_generator(G::GAPGroup)
+  @req is_cyclic(G) "the group is not cyclic"
+  length(gens(G)) == 1 && return gen(G,1)
+  return group_element(G, GAPWrap.MinimalGeneratingSet(G.X)[1])
```

Comment:
Can we also get a `minimal_generating_set` function (not in this PR, but in a follow-up perhaps)?

Reply:
Is GAP's `MinimalGeneraingSet` the right thing to ask for? Often it is sufficient to have a generating set of minimal length.